packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/geo/UnmanagedGeoPolygon.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.internal.geo
+
+import io.realm.kotlin.annotations.ExperimentalGeoSpatialApi
+import io.realm.kotlin.types.geo.GeoPoint
+import io.realm.kotlin.types.geo.GeoPolygon
+
+@OptIn(ExperimentalGeoSpatialApi::class)
+public class UnmanagedGeoPolygon(
+    public override val outerRing: List<GeoPoint>,
+    public override val holes: List<List<GeoPoint>> = emptyList()
+) : GeoPolygon {
+
+    private companion object {
+        private const val MIN_RING_SIZE = 3
+    }
+
+    init {
+        // Do basic input validation. Core will validate the rest when the query is run.
+        // Better input validation will be available later.
+        if (outerRing.size < MIN_RING_SIZE) {
+            throw IllegalArgumentException("The outer ring requires at least $MIN_RING_SIZE points: ${outerRing.size}")
+        }
+        if (outerRing.first() != outerRing.last()) {
```

Comment:
Hmm. Maybe I was not right in suggesting this check. I am a bit confused, though, so don't know what is up and down 🙃. Seems like the general `GeoPolygon` class documentation state that there must be four https://github.com/realm/realm-kotlin/pull/1403/files#diff-bb869c11647feb614648b9cdafb62fb3aa3b95c2b18f6d81f3cd714f90cce286R29-R31, but then it is also mentioned that the first and last point is implicitly connected in https://github.com/realm/realm-kotlin/pull/1403/files#diff-bb869c11647feb614648b9cdafb62fb3aa3b95c2b18f6d81f3cd714f90cce286R102-R103. The reason why I suggested the check was just to catch errors before performing the actual queries, but if there is an option to leave out the duplicate first/last point and consider it being implicit I would vote for allowing that.

Reply:
This check is totally valid as it is required that the first and last points to be the same, otherwise core would throw an exception.